bvbabel/vmr.py

```diff
@@ -50,8 +50,12 @@ def read_vmr(filename):
         # coregistration routines as well as for proper visualization.
 
         # Expected binary data: unsigned short int (2 bytes)
-        data, = struct.unpack('<H', f.read(2))
-        header["File version"] = data
+        if filename[-3:] == 'vmr':   # check file format used
```

Comment:
Hmm, I don't like that `read_vmr` function can also read other formats, it complicates the read write flow. I see that this is a quick solution but I prefer having a separate e.g. `read_v16` function for the long term maintenance and simpler read-write flow. 

You can consider writing a new separate function or I can do this. What do you think?

Reply:
Will put the v16 variant in a separate function, agree that this would be more structured and easy to read (the main reason I haven't so far was because when I started the v16 support I didn't know it would entail too many changes).

Putting it in a new function will also probably resolve the other concerns you mentioned for the other parts of the PR.

Though then still to decide whether these functions should live within the `vmr.py` file or have its own `v16.py` file. There is something to be said for both; on the one hand, v16 files are vmr files with more bits to save data (and the header cut off). On the other hand, for auto filling and quick scripting most people will likely just try `bvbabel.v16.read_v16` as it follows the general structure of the project. What do you think?

Reply:
Good points, lets err on the side of clarity & consistency and go for `bvbabel.v16.read_v16`,  we can also add this to the readme as v16 support. At the end, this is a different file format.

Reply:
One think we should remember in the future is that when doing e.g. vmr to nifti conversion, it might be better to check the existence of the paired v16 file and convert the higher precision values into the nifti. However, this can/should be solved with a higher level function calling `bvbabel.vmr.read_vmr` and bvbabel.v16.read_`v16` functions with a bit of flow control similar to your initial v16 implementation.